src/resource_clients/request_queue.ts

```diff
@@ -96,6 +97,113 @@ export class RequestQueueClient extends ResourceClient {
         return cast(parseDateFields(pluckData(response.data)));
     }
 
+    protected async _batchAddRequestsWithRetries(
+        requests: Array<Omit<RequestQueueClientRequestSchema, 'id'>>,
+        options: RequestQueueClientBatchAddRequestOptions = {},
+    ): Promise<RequestQueueClientBatchAddRequestsResult> {
+        ow(requests, ow.array.ofType(ow.object.partialShape({
+            id: ow.undefined,
+        })).minLength(1).maxLength(25));
+
+        ow(options, ow.object.partialShape({
+            forefront: ow.optional.boolean,
+            maxRetries: ow.optional.number.greaterThan(0),
```

Comment:
Now with the updated `maxRetries` logic, we could possibly allow setting `maxRetries` to 0 as well.